### PR TITLE
Adapt Model and Logic for CareFlow

### LINKED ISSUE #54: Adapt Parser, Logic and Model for CareFlow

### PR DESCRIPTION
**Pull request by Emrysil**
Fixes #54 
Created DrugInventory class which implements ReadOnlyDrugInventory to handle drug related operations.
Created PatientRecord class which implements ReadOnlyPatientRecord to handle patient related operations.
Created ClearCommand for both Patient and Drug.
Implemented CareFlow which dispatches patient and drug related operations to PatientRecord and DrugInventory respectively.
Implemented CareFlowModel and CareFlowModelManager. Implemented CareFlowLogic and CareFlowLogicManager. Changed code which uses the old logic/model to use our CareFlow version, there can be some omissions.
Other changes are made to avoid errors. (e.g. execute#Command now takes CareFlowModel as argument, so addressBook commands are no longer working, I commented out the @Override annotation such that the code still compiles).

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/AddCommand.java [hunk 1]
```diff
@@ -64,4 +64,9 @@ public boolean equals(Object other) {
                 || (other instanceof AddCommand // instanceof handles nulls
                 && toAdd.equals(((AddCommand) other).toAdd));
     }
+
+    @Override
+    public CommandResult execute(CareFlowModel model) throws CommandException {
```
Comment (nevinlim) [inline src/main/java/seedu/address/logic/commands/AddCommand.java:69]: Are we keeping both CareFlowModel and Model?

**Review by nevinlim — COMMENTED** [review]
Apart from the Model and CareFlowModel query, i think the rest of the changes LGTM

File: src/main/java/seedu/address/logic/commands/AddCommand.java (hunk 1 above)
Comment (cydtseng) [reply src/main/java/seedu/address/logic/commands/AddCommand.java:68]: I believe Model has been removed, to prevent confusion.

**Review by cydtseng — COMMENTED** [review]
lgtm

**Pull request merged**